homeassistant/components/ecobee/climate.py

```diff
@@ -564,26 +562,11 @@ def set_preset_mode(self, preset_mode: str) -> None:
         if self.preset_mode == PRESET_VACATION:
             self.data.ecobee.delete_vacation(self.thermostat_index, self.vacation)
 
-        if preset_mode == PRESET_AWAY:
+        if preset_mode == PRESET_AWAY_INDEFINITELY:
             self.data.ecobee.set_climate_hold(
                 self.thermostat_index, "away", "indefinite", self.hold_hours()
             )
-
-        elif preset_mode == PRESET_TEMPERATURE:
-            self.set_temp_hold(self.current_temperature)
-
-        elif preset_mode in (PRESET_HOLD_NEXT_TRANSITION, PRESET_HOLD_INDEFINITE):
```

Comment:
Removed because this code was broken, sending a duration instead of preset mode to ecobee.